### PR TITLE
Add FXIOS-9028 [Microsurvey] A11y checks

### PR DESCRIPTION
**Pull request by cyndichin**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9028)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/19931)

## :bulb: Description
Add appropriate a11y labels and identifiers
- Verified with VoiceOver, VoiceControl (Item numbers), and Dynamic Text

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [x] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Strings.swift
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Strings.swift:1277]: @thatswinnie I tag this v129 to avoid confusion with the string freeze, but should I wait until v128 nightly is created or does it not matter. 

cc: @adudenamedruby

File: firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:53]: Is there a reason we use two different sizes for icons instead of scaling the icon using `UIFontMetrics.default.scaledValue(for: )`?

File: firefox-ios/Client/Frontend/Strings.swift
Comment (adudenamedruby) [reply firefox-ios/Client/Frontend/Strings.swift:1277]: we should be adding strings versions with the tag of the version in which the string will be translated. THat will allow us to more easily change the `key` value as we move forward, if a change is needed.

Comment (cyndichin) [reply firefox-ios/Client/Frontend/Strings.swift:1277]: okay, so `v128` is okay as long as we're okay with it not being translated in time. will update to `v128`!

**Comment by cyndichin** [comment]
@tusharC95 @DanielDervishi

File: firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift]: thanks for calling this out @thatswinnie! Let me confirm with design on this and see if any issues, in the meantime I updated the code (can always revert).

File: firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:60]: What's the reasoning behind making the logo an a11y element? I thought that only images that convey information should be a11y elements - I don't think this logo counts as that.

File: firefox-ios/Client/Frontend/Microsurvey/Survey/View/MicrosurveyTableViewCell.swift
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Microsurvey/Survey/View/MicrosurveyTableViewCell.swift:60]: This should probably be `UnselectedRadioButtonAccessibilityLabel` here 😉

File: firefox-ios/Client/Frontend/Microsurvey/Survey/View/MicrosurveyTableViewCell.swift
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Microsurvey/Survey/View/MicrosurveyTableViewCell.swift:74]: Nit: There is one space too much before the `=`

File: firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift
Comment (thatswinnie) [inline firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:65]: Not a comment for this line but for the `titleLabel` (line 68): Could you please add `label.accessibilityTraits.insert(.header)` for the prompt title?

File: firefox-ios/Client/Frontend/Microsurvey/Survey/View/MicrosurveyTableViewCell.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/Microsurvey/Survey/View/MicrosurveyTableViewCell.swift:60]: ah, nice catch! thank you!

File: firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:60]: It conveys that this prompt is belonging to Firefox, which was a concern originally from design perspective. The note attached was `Indicates to user that prompt is coming from Firefox and not website.` While the other images in the survey are more presentational, so didn't add a11y for that.

Comment (cyndichin) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:60]: If this is still concern, I can follow up with a11y on this.

**Comment by cyndichin** [comment]
thanks for your review and your eagle eye @thatswinnie! 🥳  I addressed some of your comments and responded to the ones that were more ambiguous.

File: firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift
Comment (thatswinnie) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:65]: I can't speak to the usual arguments but tell you why I'd add it here:

1. A voice over user would have to navigate the whole website to get to the prompt - even if they are using the rotor set to headings.
2. The other to views following the prompt have one or more headings and the label in the same position is always a heading in these views.

File: firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:60]: ok great thank you!

File: firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:65]: okay, let me follow up and will report back!

Comment (cyndichin) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:65]: @thatswinnie to clarify on point 1, I was under the assumption that the `layoutChanged` navigates to the prompt regardless if its a header or not as long as we pass in the UI element as an argument to that notification. Please correct me if I am missing something.

Comment (thatswinnie) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:65]: @cyndichin you are absolutely right. Thanks for clarifying. I had previously only tested with turning voiceover on after the app was launched which lead to having trouble getting to the prompt. If I have voiceover turned on before app launch the prompt headline is getting selected through the notification.

Comment (cyndichin) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift:65]: great, thank you for checking!

File: firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift
Comment (cyndichin) [reply firefox-ios/Client/Frontend/Microsurvey/Prompt/MicrosurveyPromptView.swift]: @thatswinnie design says this is good to scale like this, thank you for pointing this out. Will keep in mind to discuss using this when scaling images for a11y.

**Comment by cyndichin** [comment]
thanks for the initial review @thatswinnie, its ready for another look! I updated the strings to `v129` since today is merge day and can wait until after the version is bump to merge this.

**Review by thatswinnie — APPROVED** [review]

**Pull request merged**
